File: code/Java/strategy/duck/RedHeadDuck.java
package strategy.duck;

import strategy.fly.FlyWithWings;
import strategy.quack.Quack;

public class RedHeadDuck extends Duck{

    public RedHeadDuck() {
        flyBehavior = new FlyWithWings();
        quackBehavior = new Quack();
    }

    @Override
    public void display() {
        System.out.println("RedHeadDuck");
    }
}

File: code/C++/strategy/redheadduck.cpp
#include "redheadduck.h"

RedHeadDuck::RedHeadDuck() {
	this->flyBehavior = new FlyWithWings();
	this->quackBehavior = new Quack();
}

void RedHeadDuck::dispaly() {
	std::cout << "RedHeadDuck" << std::endl;
}